napari/_qt/layer_controls/qt_layer_controls_base.py

```diff
@@ -148,8 +310,40 @@ def _on_ndisplay_changed(self) -> None:
         """Respond to a change to the number of dimensions displayed in the viewer.
 
         This is needed because some layer controls may have options that are specific
-        to 2D or 3D visualization only.
+        to 2D or 3D visualization only like the transform mode button.
+        """
+        self._set_transform_tool_state()
+
+    def _set_transform_tool_state(self):
         """
+        Enable/disable transform button taking into account:
+            * Layer visibility.
+            * Layer editability.
+            * Number of dimensions being displayed.
+        """
+        set_widgets_enabled_with_opacity(
+            self,
+            [self.transform_button],
+            self.layer.editable and self.layer.visible and self.ndisplay == 2,
+        )
+
+    def eventFilter(self, qobject, event):
```

Comment:
Could you add a docstring here explaining eventFilter and maybe linking to the Qt docs? I guess this is a Qt thing but I didn't know about it, so it would be nice to explain what it's doing.